PowerValidatedSolutions.psm1

```diff
@@ -50441,6 +50441,469 @@ Function Test-WMSubnetInput {
+        $formFields = @{
+            Mode = "newreq"
+            FriendlyType = "Saved-Request Certificate"
+            CertRequest = Get-Content -Path $csrFilePath -Raw
+            CertAttrib = $CertAttributes
+            TargetStoreFlags = 0
+            SaveCert = "yes"
+        }
+
+        # Request CSR to be signed by MSCA
+        Try {
+            $CertSubmitted = Invoke-WebRequest -Uri "https://$caFqdn/certsrv/certfnsh.asp" -Credential $domainCreds -Method Post -Body $formFields -SkipCertificateCheck
+            $RequestID = ($CertSubmitted.content -split '\n' | Where-Object -FilterScript {$_ -match "certnew.cer\?ReqID=[0-9]"}) -replace '[^0-9]'
+            $null = Invoke-WebRequest -Uri "https://$caFqdn/certsrv/certnew.cer?ReqID=${RequestID}&Enc=b64" -Credential $domainCreds -OutFile $crtFilePath -SkipCertificateCheck
+            if ($getCArootCert.IsPresent) {
+                $null = Invoke-WebRequest -Uri "https://$caFqdn/certsrv/certnew.p7b?ReqID=CACert&Enc=b64" -Credential $domainCreds -OutFile $rootCaFilePathTemp -SkipCertificateCheck
+                openssl pkcs7 -inform PEM -outform PEM -in $rootCaFilePathTemp -print_certs > $rootCaFilePath
+            }
+        } Catch {
+            Debug-ExceptionWriter -object $_
+        }
+
+        # Clean up
+        Remove-Item $rootCaFilePathTemp
+    } else {
+        Write-Error "Invalidate input" -ErrorAction Stop
+    }
+
+}
+Export-ModuleMember -Function Invoke-RequestSignedCertificate
+
+Function Invoke-GenerateChainPem {
+<#
+        .SYNOPSIS
+        Combine private key, signed certificate and/or root certificate chain into a single PEM file.   
+
+        .DESCRIPTION
+        The Invoke-GenerateChainPem cmdlet takes in private key, signed certificate and/or root certificate files and combine them into a single PEM file.  
+        
+        .EXAMPLE
+        Invoke-GenerateChainPem -certDirPath "C:\certificates" -keyFilePath "C:\certificates\sfo-vrli01.rainpole.io.key" -crtFilePath "C:\certificates\sfo-vrli01.rainpole.io.crt" -rootCaFilePath "C:\certificates\rpl-ad01.rainpole.io-rootCA.pem"
+        This example will combine sfo-vrli01's private key, signed certificate and rpl0-ad01's root chain certificate into single sfo-vrli01.bundle.pem file. 
+
+        .EXAMPLE
+        Invoke-GenerateChainPem -certDirPath "C:\certificates" -crtFilePath "C:\certificates\sfo-vrli01.rainpole.io.crt" -rootCaFilePath "C:\certificates\rpl-ad01.rainpole.io-rootCA.pem"
+        This example will combine sfo-vrli01's signed certificate and rpl0-ad01's root chain certificate into single sfo-vrli01.bundle.pem file. 
+
+        .EXAMPLE
+        Invoke-GenerateChainPem -certDirPath "C:\certificates" -keyFilePath "C:\certificates\sfo-vrli01.rainpole.io.key" -crtFilePath "C:\certificates\sfo-vrli01.rainpole.io.crt"
+        This example will combine sfo-vrli01's private key and signed certificate into single sfo-vrli01.bundle.pem file. 
+
+        .PARAMETER certDirPath
```

Comment:
🤔 Would `outputPath` be simpler and less confusing when seen against the other input paths?